sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -97,7 +124,18 @@ private ISpan createChild(
     Objects.requireNonNull(parentSpanId, "parentSpanId is required");
     Objects.requireNonNull(operation, "operation is required");
     final Span span =
-        new Span(root.getTraceId(), parentSpanId, this, operation, this.hub, timestamp);
+        new Span(
+            root.getTraceId(),
+            parentSpanId,
+            this,
+            operation,
+            this.hub,
+            timestamp,
+            __ -> {
+              if (waitForChildren && hasAllChildrenFinished()) {
+                finish(finishStatus);
+              }
```

Comment:
should you just call `finish(x)` directly? the `finish(x)` method does pretty much the same 2 checks

Reply:
Refactored. Take another look please.